### PR TITLE
Coadhale metrics sent to Datadog with JMX enabled

### PR DESCRIPTION
**Pull request by DhrubajyotiSadhu**

### CODE REVIEW

File: files/startup.sh [hunk 1]
```diff
@@ -86,14 +86,18 @@ fi
...
     if [ -n "$ECS_CONTAINER_METADATA_URI" ]; then
+        # This line enables JMX (Java Management Extensions) on Hadoop clients, allowing for remote monitoring and management of JVM-related metrics and operations.
+        # source: https://docs.datadoghq.com/integrations/hive/?tab=containerized
+        export HADOOP_CLIENT_OPTS="$HADOOP_CLIENT_OPTS -Dcom.sun.management.jmxremote -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.port=8088"
 
-        # enable prometheus jmx agent when running on ECS
+        # This line configures the JMX Prometheus exporter, enabling the collection of JMX metrics from the JVM and their exposure in Prometheus format for integration with monitoring systems.
         export EXPORTER_OPTS="-javaagent:/usr/lib/apiary/jmx_prometheus_javaagent-${EXPORTER_VERSION}.jar=8080:/etc/hive/conf/jmx-exporter.yaml"
+
+        # Enable it to send metrics to Cloudwatch
```
Comment (patduin) [inline files/startup.sh:97]: remove this line, it very unclear what it means.

File: files/startup.sh [hunk 2]
```diff
@@ -86,14 +86,18 @@ fi
...
         export CLOUDWATCH_NAMESPACE="${INSTANCE_NAME}-metastore"
         export ECS_TASK_ID=$(wget -q -O - ${ECS_CONTAINER_METADATA_URI}/task|jq -r .TaskARN|awk -F/ '{ print $NF }')
-        update_property.py hive.service.metrics.class com.expediagroup.apiary.extensions.metastore.metrics.CodahaleMetrics /etc/hive/conf/hive-site.xml
+        #update_property.py hive.service.metrics.class com.expediagroup.apiary.extensions.metastore.metrics.CodahaleMetrics /etc/hive/conf/hive-site.xml
```
Comment (patduin) [inline files/startup.sh:100]: Do you need those exports?
We should either:
1) remove this so we just don't support cloudwatch anymore. Make that very clear in the CHANGELOG and increase major version.
2) Make this option through some switch.

I'm ok with 1) the prometheus seems to be an ok alternative and it is less hassle to maintain two ways of reporting metrics and more in line with other Apiary projects. We can always bring it back if this breaks any users data and they can't migrate (or stay on older version).

File: files/startup.sh (hunk 1 above)
Comment (DhrubajyotiSadhu) [reply files/startup.sh]: done

File: files/startup.sh (hunk 2 above)
Comment (DhrubajyotiSadhu) [reply files/startup.sh]: Removed cloud watch completely

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html). 
 
+## [3.1.0] - 2024-02-06
```
Comment (patduin) [inline CHANGELOG.md:6]: Major version change I would say, if anyone is relying on Cloudwatch metrics this version will break potential alerting and observability.
```
## [4.0.0] - 2024-02-06
### Added
- Enables JMX (Java Management Extensions) on Hadoop clients, allowing for remote monitoring and management of JVM-related metrics 
### Removed
- CloudWatch metrics in favour of JMX Prometheus Exporter.
```

Can you explain briefly in this PR what the problem was with the Cloudwatch metrics?

Comment (DhrubajyotiSadhu) [reply CHANGELOG.md]: To send Cloudwatch metrics to Datadog we had to set up a Kinesis Data Firehose. And the metrics that were populating in Cloudwatch did not have the p90, or p50 latency metrics at all. 
It did not have many Coadhale metrics populating. 

Even we had to maintain another repo called apiary-extensions to send metrics to Cloudwatch. It was not a simple plug-and-play.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,12 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html). 
 
+## [4.0.0] - 2024-02-06
```
Comment (jmnunezizu) [inline CHANGELOG.md:6]: The `README.md` has a reference to Cloudwatch in `ENABLE_METRICS`. This env variable needs to be updated to remove such a reference.

**Review by jmnunezizu — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (DhrubajyotiSadhu) [reply CHANGELOG.md:6]: I have updated the definition of `ENABLE_METRICS` in the `README.md`

**Review by patduin — APPROVED** [review]

**Review by jmnunezizu — APPROVED** [review]

**Pull request merged**